### PR TITLE
[WFLY-19339] If Naming subsystem is activated then add both jdk.naming.dns & jdk.naming.rmi dependencies to deployments

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-19339

### CODE REVIEW

File: naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        final ModuleSpecification moduleSpecification = deploymentUnit.getAttachment(Attachments.MODULE_SPECIFICATION);
+        final ModuleLoader moduleLoader = Module.getBootModuleLoader();
+        for (String moduleName : JDK_NAMING_MODULES) {
+            try {
+                moduleLoader.loadModule(moduleName);
+                moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, moduleName, false, false, false, false));
```
Comment (bstansberry) [inline naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java:38]: FYI:

https://issues.redhat.com/browse/WFCORE-6889

File: naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            } catch (ModuleLoadException ex) {
+                NamingLogger.ROOT_LOGGER.debugf("Module not found: %s", moduleName);
```
Comment (bstansberry) [inline naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java:40]: I was thinking maybe log this at at INFO (with an i18n message) but I think a user would have to have gone far out of their way to have one of these modules be unavailable. So debug seems ok.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/naming/dns/DnsContextLookupBean.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+@Stateless
+public class DnsContextLookupBean {
+
+    public void testDnsContextLookup(String serverAddress) throws Exception {
+        final Hashtable<String, String> env = new Hashtable<>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.dns.DnsContextFactory");
+        env.put(Context.PROVIDER_URL, "dns://" + serverAddress);
+
+        Context ctx = new InitialContext(env);
+        ctx.close();
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/naming/dns/DnsContextLookupBean.java:26]: The fails with the SecurityManager enabled; needs some permission.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/naming/dns/DnsContextLookupBean.java (hunk 3 above)
Comment (ropalka) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/naming/dns/DnsContextLookupBean.java:26]: Missing permission was added.

File: naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java (hunk 1 above)
Comment (ropalka) [reply naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java:38]: OK

File: naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java (hunk 2 above)
Comment (ropalka) [reply naming/src/main/java/org/jboss/as/naming/deployment/JdkDependenciesProcessor.java:40]: OK

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @ropalka

**Pull request merged**